include/flatbuffers/vector_downward.h

```diff
@@ -139,17 +142,18 @@ class vector_downward {
   }
 
   inline uint8_t *make_space(size_t len) {
-    size_t space = ensure_space(len);
-    cur_ -= space;
+    if (len) {
```

Comment:
Is this called with `0` enough that the `if` is worth it? Can we instead reduce such callers?

Reply:
I think it's called a lot when it attempts to pad/align something and it doesn't need to since it is already aligned.

Reply:
Also, it is probably the `ensure_space` call that is the reason it is slow, since that is doing multiple assertions and conditional checks that are invariant if len == 0.

Reply:
worth checking if 90% of the `0` callers come from align/pad and then moving the if check there?

Reply:
I tried pushing the conditional up the stack a bit, an while it is better than the base case, it isn't better than doing the conditional in `make_space`. Its less code too so I would say this is the way to go for now.